code/Model/EditableFormField/EditableFileField.php

```diff
@@ -221,14 +238,22 @@ public function getFormField()
         }
 
         $folder = $this->Folder();
+        
+        if ($protectFolder && $folder->ID === 0) {
+            Versioned::set_stage($this->config()->get('file_upload_stage'));
+
+            $folderName = sprintf('%s_%s', $this->ParentID, $this->ID);
+            $folder = UserDefinedFormAdmin::getFormSubmissionFolder($folderName);
+            $this->FolderID = $folder->ID;
+            $this->write();
```

Comment:
```suggestion
        Versioned::withVersionedMode(function () {
            Versioned::set_stage($this->config()->get('file_upload_stage'));
            $folderName = sprintf('%s_%s', $this->ParentID, $this->ID);
            $folder = UserDefinedFormAdmin::getFormSubmissionFolder($folderName);
            $this->FolderID = $folder->ID;
            $this->write();
        });
```

Wrapping this block in `Versioned::withVersionedMode` means it will reset the versioned mode back to what it was before